Client/Frontend/Settings/NotificationsSettingsViewController.swift

```diff
@@ -0,0 +1,132 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0
+
+import Foundation
+import Shared
+
+class NotificationsSettingsViewController: SettingsTableViewController, FeatureFlaggable {
+    lazy var allowAllNotifications: BoolSetting = {
+        return BoolSetting(
+            title: .Settings.Notifications.AllowAllNotificationsTitle,
+            prefs: prefs,
+            prefKey: PrefsKeys.Notifications.AllowAllNotifications
+        ) { [weak self] value in
+            guard let self = self else { return }
+
+            Task {
+                let shouldEnable = await self.notificationsChanged(value)
+                self.allowAllNotifications.control.setOn(shouldEnable, animated: true)
+                self.allowAllNotifications.writeBool(self.allowAllNotifications.control)
+
+                self.tabsNotifications.control.setOn(shouldEnable, animated: true)
+                self.tabsNotifications.writeBool(self.tabsNotifications.control)
+
+                self.syncSignInNotifications.control.setOn(shouldEnable, animated: true)
+                self.syncSignInNotifications.writeBool(self.syncSignInNotifications.control)
+
+                self.tipsAndFeaturesNotifications.control.setOn(shouldEnable, animated: true)
+                self.tipsAndFeaturesNotifications.writeBool(self.tipsAndFeaturesNotifications.control)
+
+                self.tabsNotifications.enabled = shouldEnable
+                self.syncSignInNotifications.enabled = shouldEnable
+                self.tipsAndFeaturesNotifications.enabled = shouldEnable
+
+                self.tableView.reloadData()
+            }
+        }
+    }()
+
+    lazy var tabsNotifications: BoolSetting = {
+        let enabled = prefs.boolForKey(PrefsKeys.Notifications.AllowAllNotifications) ?? false
+        return BoolSetting(
+            title: .Settings.Notifications.TabsNotificationsTitle,
+            description: .Settings.Notifications.TabsNotificationsStatus,
+            prefs: prefs,
+            prefKey: PrefsKeys.Notifications.TabsNotifications,
+            enabled: enabled
+        ) { value in
+            print("tabsNotifications is \(value)")
```

Comment:
I'm not sure we should be leaving random print statements in the code. Are these useful? If so, then we should be putting them into the logs - see the [docs for our logging strategy](https://github.com/mozilla-mobile/firefox-ios/wiki/Logging-Strategy). Otherwise, let's not pollute our console with stuff we don't really need. I'm open to arguments being made for this though.

Reply:
Certainly not. There should be the code that tells the Notification server to allow for the specific notification case to be enabled or disabled. I'm not sure how to do that yet and I forgot to remove the prints. @thatswinnie is there un update on that?